src/sentry/integrations/msteams/webhook.py

```diff
@@ -89,25 +94,87 @@ def post(self, request):
         verify_signature(request)
 
         data = request.data
+        # only care about conversationUpdate
+        if data["type"] != "conversationUpdate":
+            return self.respond(status=204)
+
         channel_data = data["channelData"]
         event = channel_data.get("eventType")
+
         # TODO: Handle other events
         if event == "teamMemberAdded":
-            # only care if our bot is the new member added
-            matches = filter(lambda x: x["id"] == data["recipient"]["id"], data["membersAdded"])
-            if matches:
-                team_id = channel_data["team"]["id"]
+            return self.handle_member_added(request)
+        elif event == "teamMemberRemoved":
+            return self.handle_member_removed(request)
+
+        return self.respond(status=204)
+
+    def handle_member_added(self, request):
+        data = request.data
+        channel_data = data["channelData"]
+        # only care if our bot is the new member added
+        matches = filter(lambda x: x["id"] == data["recipient"]["id"], data["membersAdded"])
+        if not matches:
+            return self.respond(status=204)
+
+        team = channel_data["team"]
 
-                access_token = get_token_data()["access_token"]
+        # TODO: add try/except for request exceptions
+        access_token = get_token_data()["access_token"]
 
-                # need to keep track of the service url since we won't get it later
-                signed_data = {"team_id": team_id, "service_url": data["serviceUrl"]}
+        # need to keep track of the service url since we won't get it later
+        signed_data = {
+            "team_id": team["id"],
+            "team_name": team["name"],
+            "service_url": data["serviceUrl"],
+            "expiration_time": int(time.time()) + INSTALL_EXPIRATION_TIME,
+        }
 
-                # sign the params so this can't be forged
-                signed_params = sign(**signed_data)
+        # sign the params so this can't be forged
+        signed_params = sign(**signed_data)
```

Comment:
Is this creating a JWT?

Reply:
ah nvm. I looked at the tests.

Reply:
@NisanthanNanthakumar Good question. It's not using JWT but I'm not actually sure what the name of this encryption is called from the docs: https://docs.djangoproject.com/en/1.10/topics/signing/

Reply:
@scefali interesting! but we are using `@patch("jwt.decode")` in the tests?

Reply:
@NisanthanNanthakumar The JWT token that's involved is in the header of the request that we verify. This signed parameter is something we generate for the link we send to users. Two different things going on here :)